.github/workflows/ci.yml

```diff
@@ -1,58 +1,47 @@
-name: CI
+name: Test and release when tagged
 
-on: [push, pull_request]
+on: [push]
 
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: prefix-dev/setup-pixi@v0.8.1
+        with:
+          environments: lint
+      - run: pixi run --environment lint lint
+
   test-linux:
+    runs-on: ubuntu-latest
     strategy:
-      fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        environment:
+          - test-py39
+          - test-py310
+          - test-py311
+          - test-py312
+          - test-py313
         numpy: [null, "numpy>=1.23,<2.0.0", "numpy>=2.0.0rc1"]
         uncertainties: [null, "uncertainties==3.1.6", "uncertainties>=3.1.6,<4.0.0"]
```

Comment:
Dunno if its still relevant to test multiple versions of uncertainties

Reply:
not sure. My suggestion is that we use this commit to migrate as is, and the use future commits to do this.